src/common/storage/src/stage.rs

```diff
@@ -107,9 +107,15 @@ impl StageFilesInfo {
         }
     }
 
-    pub async fn list(&self, operator: &Operator, first_only: bool) -> Result<Vec<StageFileInfo>> {
+    pub async fn list(
+        &self,
+        operator: &Operator,
+        first_only: bool,
+        file_limit: usize,
```

Comment:
should not limit here,  some may already be copied.
limit should be applied when filtering colored files.

Reply:
It's hard to track, I think the users should use `purge = 1` to make it right.  So it's reasonable to use limit here.